sky/clouds/service_catalog/common.py

```diff
@@ -503,18 +503,21 @@ def list_accelerators_impl(
     def make_list_from_df(rows):
         if all_regions:
             # Keep all regions.
-            rows = rows.groupby([
-                'InstanceType', 'AcceleratorName', 'AcceleratorCount', 'vCPUs',
-                'MemoryGiB', 'Region'
-            ],
-                                dropna=False).aggregate('min').reset_index()
+            rows = rows.sort_values(
+                by=['Price', 'SpotPrice', 'Region']).drop_duplicates(
+                    subset=[
+                        'InstanceType', 'AcceleratorName', 'AcceleratorCount',
+                        'vCPUs', 'MemoryGiB', 'Region'
+                    ],
+                    keep='first')
         else:
             # Only keep the lowest prices across regions.
-            rows = rows.groupby([
-                'InstanceType', 'AcceleratorName', 'AcceleratorCount', 'vCPUs',
-                'MemoryGiB'
-            ],
-                                dropna=False).aggregate('min').reset_index()
+            rows = rows.sort_values(by=['Price', 'SpotPrice']).drop_duplicates(
+                subset=[
+                    'InstanceType', 'AcceleratorName', 'AcceleratorCount',
+                    'vCPUs', 'MemoryGiB'
+                ],
```

Comment:
Just a nit: may be we can simplify the code and reduce the duplication by something like the following:
```python
        sort_key = ['Price', 'SpotPrice']
        subset = ['InstanceType', 'AcceleratorName', 'AcceleratorCount',
                        'vCPUs', 'MemoryGiB']
        if all_regions:
            sort_key.append('Region')
            subset.append('Region')

        rows = rows.sort_values(by=sort_key).drop_duplicates(
                    subset=subset,
                    keep='first')
```

Reply:
done